lib/experimental/class-wp-html-walker-test.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+
+require_once './class-wp-html-walker.php';
+
+final class WP_HTML_Walker_Tests extends TestCase {
+	const HTML_SIMPLE = '<div id="first"><span id="second">Text</span></div>';
+	const HTML_WITH_CLASSES = '<div class="main with-border" id="first"><span class="not-main bold with-border" id="second">Text</span></div>';
+
+	public function test_to_string_with_no_updates_returns_the_original_html() {
+		$w = new WP_HTML_Walker( self::HTML_SIMPLE );
+		$this->assertSame( self::HTML_SIMPLE, $w . '' );
```

Comment:
Type casting is easier to read in PHP:
```suggestion
		$this->assertSame( self::HTML_SIMPLE, (string) $w );
```

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/42485/commits/5bf0c2815785c2a7f91fe844aa82cc0cafb3f5af